src/HealthGPS.Console/model_parser.cpp

```diff
@@ -101,10 +102,12 @@ HierarchicalLinearModelDefinition load_static_risk_model_definition(const host::
                               .variances = at.variances});
     }
 
-    return HierarchicalLinearModelDefinition{std::move(models), std::move(levels)};
+    return std::make_shared<HierarchicalLinearModelDefinition>(std::move(models),
+                                                               std::move(levels));
 }
 
-LiteHierarchicalModelDefinition load_dynamic_risk_model_definition(const host::poco::json &opt) {
+std::shared_ptr<LiteHierarchicalModelDefinition>
+load_dynamic_risk_model_definition(const host::poco::json &opt) {
```

Comment:
Can you give me a 1-line explanation of what the purpose of using `std::make_shared` and `std::shared_ptr`. It seems like an important change in the overall architecture of the models, but as I'm new to C++, I cannot grasp it by just reading the code or googling the answer.

Reply:
I use `std::shared_ptr`s here to allow the vector of model definitions (`shared_ptr`s to base model definition class objects) to hold derived model definition class objects and polymorphically call the correct `create_model`.

Not quite a one-liner any more, but `shared_ptr` is required over a `unique_ptr` because `shared_ptr` allows us to copy it from the storage vector.